recipes/absent/all/conanfile.py

```diff
@@ -19,36 +19,29 @@ class AbsentConan(ConanFile):
     def _source_subfolder(self):
         return "source_subfolder"
 
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
-    
     def _supports_cpp17(self):
         supported_compilers = [("gcc", "7"), ("clang", "5"), ("apple-clang", "10"), ("Visual Studio", "15.7")]
         compiler = self.settings.compiler
-        version = Version(self.settings.compiler.version)
-        return any(compiler == e[0] and version >= e[1] for e in supported_compilers)
+        version = Version(compiler.version)
+        return any(compiler == sc[0] and version >= sc[1] for sc in supported_compilers)
             
     def _configure_cmake(self):
         cmake = CMake(self)
         cmake.definitions["BUILD_TESTS"] = "OFF"
-        cmake.configure(source_folder=self._source_subfolder,
-                        build_folder=self._build_subfolder)
+        cmake.configure(source_folder=self._source_subfolder)
         return cmake
 
     def configure(self):
-        if not self._supports_cpp17():
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "17")
+        elif not self._supports_cpp17():
             raise ConanInvalidConfiguration("Absent requires C++17 support")
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         extracted_dir = self.name + "-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
 
-    def build(self):
-        cmake = self._configure_cmake()
```

Comment:
why remove build?

Reply:
Because it's a header-only library. So, I thought that the build step was unnecessary. Should I bring it back?

Reply:
no, I think this is not needed in this case. I wonder why did we have it all.